runtime/fastly/host-api/host_api_fastly.h

```diff
@@ -107,11 +108,33 @@ class FastlyAsyncTask final : public AsyncTask {
       abort();
     handle_ = static_cast<int32_t>(handle);
   }
+  FastlyAsyncTask(uint32_t handle, JSContext *cx, JS::HandleObject context,
+                  JS::HandleObject promise, ProcessAsyncTask *process) {
+    if (static_cast<int32_t>(handle) < 0)
```

Comment:
Shouldn't this be a comparison against the `INVALID_HANDLE` constant? The conversion to a signed type is a little concerning, and I'd be more comfortable with a direct comparison against the known invalid handle value (if that's what this conditional is implementing).

Reply:
The StarlingMonkey handles are signed ints https://github.com/bytecodealliance/StarlingMonkey/blob/main/include/extension-api.h#L27, so since we are casting our Fastly handles, it seemed like asserting we never go into the signed range would be the stronger assertion. We could just make this an invalid handle assertion though, yes.

Reply:
I think it's reasonable to check that it's always greater than zero, but it might violate assumptions that compute makes about the valid range of handles. I'm reasonably sure we'll never see handles that have the top-bit set, but it would be easiest to assume that we only care about the invalid handle sentinel.